fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/IndexKeyValueToPartialRecord.java

```diff
@@ -298,16 +303,20 @@ public void addRequiredMessageFields() {
          * @return whether this is a valid use
          */
         public boolean isValid() {
+            return isValid(false);
+        }
+
+        public boolean isValid(boolean allowRepeated) {
```

Comment:
There's something about this where it seems like the right thing is to have a key expression or something like a key expression that means "one entry from a repeated field" to avoid having a Boolean special case, but I suppose that's a much larger discussion. cc @normen662

Reply:
I'm not sure how a new key expression might be used, since as things stand no new types are introduced.

I think the "right thing" is to have dynamic row types so that it isn't necessary to impersonate the record with varying degrees of fidelity. Maybe that's the same thing.